clients/mediaconvert.go

```diff
@@ -145,15 +200,31 @@ func (mc *MediaConvert) Transcode(ctx context.Context, args TranscodeJobArgs) er
 
 // This is the function that does the core AWS workflow for transcoding a file.
 // It expects args to be directly compatible with AWS (i.e. S3-only files).
-func (mc *MediaConvert) coreAwsTranscode(ctx context.Context, args TranscodeJobArgs, accelerated bool) error {
+func (mc *MediaConvert) coreAwsTranscode(ctx context.Context, args TranscodeJobArgs, accelerated bool) (err error) {
 	log.Log(args.RequestID, "Creating AWS MediaConvert job", "input", args.InputFile, "output", args.HLSOutputFile, "accelerated", accelerated)
-	payload := createJobPayload(args.InputFile.String(), args.HLSOutputFile.String(), mc.role, accelerated)
+
+	transcodeProfiles := args.Profiles
+	if len(transcodeProfiles) == 0 {
+		transcodeProfiles, err = video.GetPlaybackProfiles(args.InputFileInfo)
+		if err != nil {
+			return fmt.Errorf("failed to get playback profiles: %w", err)
+		}
+	}
+
+	hlsOut := args.HLSOutputFile.String()
+	mp4Out := ""
+	if args.InputFileInfo.Duration <= maxMP4OutDuration.Seconds() {
+		pathParts := strings.Split(hlsOut, string(filepath.Separator))
+		pathParts[len(pathParts)-1] = "static"
```

Comment:
@thomshutt are you happy with us deciding the name of the mp4 outs and just putting them in the same path as the hls outputs?

Reply:
Yep, that should be fine